sources/host-ctr/cmd/host-ctr/main.go

```diff
@@ -942,6 +979,61 @@ func withDynamicResolver(ctx context.Context, ref string) containerd.RemoteOpt {
 		}
 	default:
 		// For all other registries
-		return noOp
+		return defaultResolver
+	}
+}
+
+// registryHosts returns the registry hosts to be used by the resolver. Wraps around _registryHosts.
+// Heavily borrowed from containerd CRI plugin's implementation with the auth related configuration omitted.
+// See https://github.com/containerd/cri/blob/f6026296a3991010429db91e7e677f9c9d4861ab/pkg/server/image_pull.go#L314-L315
+// FIXME Replace this once there's a public containerd client interface that supports registry mirrors
+func registryHosts(registryConfig *RegistryConfig, authorizer docker.Authorizer) docker.RegistryHosts {
+	return func(host string) ([]docker.RegistryHost, error) {
+		return _registryHosts(host, registryConfig, authorizer)
+	}
+}
+
+// Separated out from registryHosts so it is unit-testable. (assert.Equal() does not work on function pointers)
```

Comment:
nit: this doesn't _have_ to be separated as the test could do something like this:

```go
f := registryHosts(config, authorizer)
result, err := f(host, config, authorizer)
```